bigquery/src/test/java/org/apache/iceberg/gcp/bigquery/TestBigQueryCatalog.java

```diff
@@ -0,0 +1,673 @@
+        // Namespace might be dropped by a previous iteration, ignore
+      }
+    }
+  }
+
+  @Override
+  protected boolean requiresNamespaceCreate() {
+    return true;
+  }
+
+  @Override
+  protected boolean supportsNamesWithSlashes() {
+    return false;
+  }
+
+  @Override
+  protected boolean supportsNamesWithDot() {
+    return false;
+  }
+
+  @Override
+  public BigQueryMetastoreCatalog catalog() {
+    return catalog;
+  }
+
+  @Override
+  protected BigQueryMetastoreCatalog initCatalog(
+      String catalogName, Map<String, String> additionalProperties) {
+
+    warehouseLocation = tempFolder.toPath().resolve("hive-warehouse").toString();
+
+    Map<String, String> properties =
+        Map.of(
+            ICEBERG_CATALOG_TYPE,
+            ICEBERG_CATALOG_TYPE_BIGQUERY,
+            PROJECT_ID,
+            GCP_PROJECT,
+            CatalogProperties.WAREHOUSE_LOCATION,
+            warehouseLocation,
+            TESTING_ENABLED,
+            "true",
+            CatalogProperties.TABLE_DEFAULT_PREFIX + "default-key1",
+            "catalog-default-key1",
+            CatalogProperties.TABLE_DEFAULT_PREFIX + "default-key2",
+            "catalog-default-key2",
+            CatalogProperties.TABLE_DEFAULT_PREFIX + "override-key3",
+            "catalog-default-key3",
+            CatalogProperties.TABLE_OVERRIDE_PREFIX + "override-key3",
+            "catalog-override-key3",
+            CatalogProperties.TABLE_OVERRIDE_PREFIX + "override-key4",
+            "catalog-override-key4");
+
+    BigQueryMetastoreCatalog tmpCatalog =
+        (BigQueryMetastoreCatalog)
+            CatalogUtil.buildIcebergCatalog(
+                catalogName,
+                ImmutableMap.<String, String>builder()
+                    .putAll(properties)
+                    .putAll(additionalProperties)
+                    .build(),
+                null);
+
+    return tmpCatalog;
+  }
+
+  @Test
+  public void testRenameTable() {
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(NS);
+    }
+
+    assertThat(catalog.tableExists(TABLE))
+        .as("Source table should not exist before create")
+        .isFalse();
+
+    catalog.buildTable(TABLE, SCHEMA).create();
+    assertThat(catalog.tableExists(TABLE)).as("Table should exist after create").isTrue();
+
+    assertThat(catalog.tableExists(RENAMED_TABLE))
+        .as("Destination table should not exist before rename")
+        .isFalse();
+
+    assertThatThrownBy(() -> catalog.renameTable(TABLE, RENAMED_TABLE))
+        .isInstanceOf(UnsupportedOperationException.class)
+        .hasMessageContaining("Table rename operation is unsupported");
+
+    catalog.dropTable(TABLE);
+  }
+
+  @Disabled(
+      "BigQuery does not allow adding a REQUIRED column to an existing table schema. "
+          + "When this operation is attempted, an exception gets thrown.")
+  @Test
+  public void testOverrideTablePropertiesReplaceTransaction() {}
```

Comment:
is this still true? when I removed the `Disabled` flag the test would pass

Reply:
given that the tests here that live in the Iceberg codebase are always running against the Fake client, I would suggest to remove all of those `Disabled` tests, since they all pass against the Fake client. If you want to run the catalog tests against an actual client, you can then basically just create a new `TestRealBigQueryCatalog` that initializes the correct client where you disable the tests that don't work

Reply:
Done